pkg/ipam/address/address.go

```diff
@@ -74,6 +74,18 @@ type ReserveRandom struct {
 	LocationID string `json:"location_identifier"`
 	VlanID     string `json:"vlan_identifier"`
 	Count      int    `json:"count"`
+
+	// PrefixID limits the potential addresses to a specific prefix
+	// within the specified VLAN. Defaults to all prefixes within the VLAN.
+	PrefixID string `json:"prefix_identifier,omitempty"`
+
+	// IPVersion limits the potential addresses to a specific IP version.
+	// Defaults to v4 & v6.
+	IPVersion int `json:"ip_version,omitempty"`
```

Comment:
So, `0` means IPv4 and IPv6, `4` means IPv4 and `6` means IPv6. That's one hell of an API (not to blame it on you, it's not your fault.)

But I think we should at least add some validation *and* better documentation to let users know what's going to happen at which point. Maybe we could even add a "enum" type for this property to have at least some self-documented code. Something along the lines of:

```go
type IPVersionLimit int

const (
	IPVersionLimitDualstack IPVersionLimit = 0
	IPVersionLimitIPv4 IPVersionLimit = 4
	IPVersionLimitIPv6 IPVersionLimit = 6 
)
```

What do you think?

Reply:
`0` is simply the zero-value which is omitted (via omitempty json option) which means that we don't have any limits on which version is requested.

An "enum" type would probably help :+1:. I think I'd leave out the `Dualstack`  value, as it might suggest that one of both types will be reserved. But it will only be one - either v4 or v6, depending on what is available. ..my comment was probably misleading. I'll try to clarify it.

Validation by the server should be sufficient, I think. But please let me know if you have a strong opinion on adding client side validation for this.